### ISSUE TITLE
sources?

### ISSUE THREAD

**Issue by zwhitchcox** [ZeWaren/python-upnp-ssdp-example#2]
Hey, I this is extremely useful, thank you so much for creating this, but there is a lot of low level programming in here, and I'm just wondering what exactly was your reference when you created this, so that maybe I can sort of trace the logic a little better, because I've never really programmed a webserver at this low of a level.

Thanks for creating this!

**Comment by ZeWaren** [comment]
Hey Zane,

In this specific example, part of the code is from an old legacy project from 10 years ago. I just took it, and extended it so that it fitted what I needed at the time using the official specifications of uPnP. It was only about following a PDF and implementing what was there.

Technically, there was no need to re-implement a HTTP server. If I had written the code myself from scratch, I would have used an existing server. This is exactly what I did when I wrote class `UPNPHTTPServerHandler`.

The only reason ssdp.py has its own HTTP server is that the code was working when I found it, so I reused it without rewriting it for the sake of rewriting it.
